### ISSUE TITLE
[FEATURE] Improve the configurability of the publication module

### ISSUE THREAD

**Issue by peer35** [UtrechtUniversity/yoda-ansible#417]
#### Is your feature request related to a problem? Please describe.
After every Yoda upgrade we run into problems with the publication module:
- DataCite Publisher is set to Utrecht University.
- The landing page templates revert to the UU versions, with hardcoded references to the UU in the HTML meta tags and a link to UU RDM support.
- The styling and logo revert to the UU versions.

As rodsadmin we can change the landing page templates but not the CSS, logo and Publisher field.

As far as I know SURF can only fix the setup manually after an upgrade. We try to supply our needed settings before the upgrade, but this workflow is prone to errors and time consuming for both the institution and SURF.

These errors also have a relatively high impact because they can result in incorrect DataCite registrations and are very visible to end-users.

#### Describe the solution you'd like
Ideally the publication settings should be configurable via a "Yoda Admin page" as in #366 and be persistent over Yoda upgrades. 

As an alternative the publication settings should be configurable by a SURF admin as config options before an install/upgrade. 

It would also be a bonus if there is an easy way (at least for the SURF admins) to do a test publication with a DataCite TEST configuration, so we do not have to wait for a researcher to publish a real dataset after an upgrade to see if everything is correct. (registering a test dataset in DataCite production is of course a no-go)

#### Describe alternatives you've considered
-

#### Additional context
-

**Comment by Danny-dK** [comment]
This would be great. Ideally you would want to have something like 'if theme is VU, then load all VU associated configurations'. That probably would mean that all required configuration files need to be available in Yoda as default to be loaded by SURF for each institute using Yoda that wants institution themes. I understand that my question regarding the required files and css files for publication module at WUR is registered internally as https://utrechtuniversity.atlassian.net/browse/YDA-5671. 

As a side note, I could also imagine that it might have been better to not allow personalization to the extend that is possible. Perhaps just one colour scheme and at the most change logo's, institute name, and email contact addresses, while the publication templates would then contain general information that would be applicable regardless of institute (most of the textual contents of those templates are generic anyway also at UU)?

**Comment by Danny-dK** [comment]
@lwesterhof  @RobvanSchip 
Regarding my email on the files required to be changed for activating the publication module in a certain theme. You indicated that a specific feature request would need to be created and discussed, which would be this feature from Peter. Discussing this with Peter, we kinda had the question why changing colours and adding a logo has to be so complicated. It would be great if there is an easier way to do so. You indicated some aspects will be integrated into an admin page https://github.com/UtrechtUniversity/yoda/issues/366. That might take some time and may not be able to configure the landing pages. It would be great in the meantime if an intermediate solution is available (such as creating a neutral colour theme for landing pages (my suggestion, not necessarily Peter's))?

**Comment by NikaVK** [comment]
This morning we have been dealing with this issue again, both VU and WUR. The landing pages after the Yoda upgrade are not working as they should and as a consequence, the publishing through Yoda is not possible for both instances. Has any progress been made on resolving this issue?

**Comment by lwesterhof** [comment]
Thank you for your update. I want to clarify that while the templating option for landing pages does exist, it hasn't been used yet. We've had discussions with the SURF team about bringing these templates under version control for future deployments. If you have any further concerns or need additional information, please let us know.

**Comment by OrlandoYoDa** [comment]
We have created internal ticket YDA-7007 for discussion.

**Comment by peer35** [comment]
Please give this high priority. The landing page templates are once again overwritten with the UU defaults during the Yoda 2.1 upgrade, creating unecessary extra work for SURF and institutional admins.

**Comment by Danny-dK** [comment]
Is it unnecessary work for SURF? SURF is the hosting institution that is being paid to uphold the service with the details of the institutions in mind? Although configurability of the publication module is highly required, would this in the meantime not be a workflow adjustment when updating? I would imagine that UU devs would be able to talk with SURF on how to handle all steps during update? In my limited and n00b view I would imagine:
- copy settings and required configuration files to a save area
- update Yoda
- make sure Yoda works
- transfer the settings and configuration files
- make sure Yoda works

My view above is likely very limited (@lwesterhof is it?), but I would imagine this to be part of the hosting organisation to make sure this is correct. Otherwise they should not have agreed to institutional configurations.

**Comment by OrlandoYoDa** [comment]
@peer35 and @Danny-dK, thank you for your input. We are reviewing your comments and updating this feature on our backlog. It remains to be seen what the best course of action might be. We'll keep you informed.

**Comment by MKoperdraad** [comment]
Hi Peter and Danny
 
Thank you for your feedback. 
 
I'd like to first comment on the immediate technical aspects of the issue:
- In Yoda 1.10.0, we've introduced landing page themes, so 
  organizations no longer need to maintain landing page templates 
  if they just want to change the look and feel of landing pages (e.g. a different logo,
  fonts, colours, etc.). Although they still need to be tested before application
  upgrades, they require less frequent updates than landing page templates since
  they only have layout information.
- There's still an ongoing stakeholder discussion about the requirements
  for configuring an alternative publisher (for example: what kind of exceptions should be supported, if any,
  what criteria they should be based on, and how all of this needs to be configured). We're currently
  waiting for a decision on requirements before we can refine and implement this request.
  If the requirements are simple, we should be able to include this soon, e.g. in the next bugfix release,
  depending on the exact timing. If requirements are more complex, this needs to be considered in roadmap
  planning.
  
 At the same time, as Danny already mentioned, these incidents have both technical
 and organizational aspects, with the organizational side generally having more impact.
 
 We can of course improve configuration functionality for landing pages, but this will have little or no effect on problems
 like the ones you mentioned without required organizational processes being in place. For a customized
 version of Yoda to be offered as a stable service, it's essential to have processes such as acceptance testing
 on a separate representative acceptance environment, and for reviewing configuration changes before applying them
 in production. If we can get such processes in place, we can achieve these goals effectively, without a large
 amount of development effort.
 
 Without these organizational processes in place, fully addressing the current request in a robust and reliable way
 would require quite large amount of stakeholder input, development and testing effort. This would also need to be
 accommodated within the product roadmap.  In this situation, we  don't believe that would be the most effective use of the available development capacity. Setting up the  organizational processes first will allow us to simplify technical improvements, so that we wouldn't have to postpone  other features to accomplish this result.
 
 If you have an opinion about the requirements for the publisher configuration (e.g. is an environment-wide configuration
 sufficient for the time being, or do you need more flexibility for handling special cases), we're interested in hearing
 them. If you are aware of important exceptions or special cases, can you share (anonymized) examples?

(btw this response is based on team discussion)_

**Comment by Danny-dK** [comment]
Thanks for the feedback. I'm not sure I understand though. Organisations have made their own adjustments to the look (colours and logo's) and text of the landingpages. SURF applies these. These are present within the software as settings / files? Every time SURF updates Yoda, everything reverts to the UU look and feel, e.g. the previous settings are overwriten. Why is it not possible for SURF to apply the colour, look, and texts (that they already have and applied) to each new update? Is it difficult to save those aspects per institute and return them after an update? As far as I know, after an update, institutes find out that the adjustments are gone, they then ask SURF to apply them again, and then they have them again. Why would the hosting institutions not be able to do:
- extract institutional adjustments to colours and logo's and text before an update.
- update Yoda
- add the institutional adjustments to colours logo's and text.

**Comment by MKoperdraad** [comment]
Hi Danny and all, personally I think a lot could be gained by having a meeting  with representatives of Yoda team, Surf and institutions, to discuss the process and mutual expectations.

**Comment by ccacciari** [comment]
SURF does create backups of the institutional adjustments before any upgrade, but that is not enough:
 
1. During the latest upgrade, the challenge was that the logic surrounding templates had been changed in such a way that the existing custom ones of institutes would no longer work without having the same updates applied. We at SURF didn't realise this until very late in the day, so we gave very little notice to VU to update their templates in time. In general, we need to implement a way of keeping on top of any changes that might affect customisation. 
2. From the FA's and hoster's perspective, the best solution for the landing pages in general would be to have the main configurable elements (e.g., colour scheme, logo, institute name..) as options in the admin portal. This of course has to be weighed up against the development effort required.

**Comment by peer35** [comment]
Looking back at my original issue, there were 3 points:
1. DataCite Publisher is set to Utrecht University.
Solved. SURF asked us what it should be and so far it persists over upgrades.

2. The landing page templates revert to the UU versions, with hardcoded references to the UU in the HTML meta tags and a link to UU RDM support.
**This happened again with the 2.1 upgrade.**

3. The styling and logo revert to the UU versions.
Solved, we have our institutional css and logo for the landing pages which persists over upgrades. 
Note: can be much improved because it is a pain to create and test the css. A Yoda functional admin shouldn't have to know HTML, CSS and how to set up a Yoda test environment. 

On the short term 2 could be solved by using variables in the templates for the institution name and support link so they can be set once by SURF and then persist over upgrades. 

We have no need to further customize the landing page templates and are happy to follow the latest versions, but our landing pages must be VU branded and include a link to our own support page.

On the longer term it would be of great benefit if institutions can simply manage "institutional defaults" like these themselves in an admin interface. That will of course mean a major overhaul of Yoda, but we would like to see that on the roadmap as improvements to the user friendliness for institutional and technical (SURF) admins.

**Comment by NikaVK** [comment]
Dear all,

I agree with Peter on his last post - landing pages must be institution branded and include links to institution's support. At the moment I also do not foresee any changes in the WUR publication landing pages, but I would like to have an option to apply any changes when they are needed.  Like Peter and Claudio said, it would be of great benefit if the the institutional defaults could be edited in the Administration panel.

**Comment by OrlandoYoDa** [comment]
Hi @peer35, @NikaVK, @Danny-dK, @ccacciari, and the other stakeholders,

Thank you for your input. Your request has been duly noted.

I believe this feature aligns well with the broader UI and usability improvements outlined in our roadmap. While implementing it will require a significant investment of development effort, I agree that it would add substantial value. I'll make sure it receives the attention it deserves when we review and prioritize upcoming work.

In the meantime, I think it would be worthwhile to review the current template update process together and see whether there are improvements we can make in the shorter term.

@ccacciari, would you be willing to organize such a meeting if you think it would be useful?

Kind regards,
Orlando

**Comment by ccacciari** [comment]
I agree, a meeting would be useful.
I will plan something.

Kind Regards,
Claudio

**Comment by MKoperdraad** [comment]
Hi Claudio, could you please invite Orlando, Claire and myself from Yodateam? Of course we hope Peter and Nika will also be there, so we'll all be on the same page. Thanks for organizing!

**Comment by OrlandoYoDa** [comment]
Hi @peer35 et al! 

The maintenance of landing page templates through the Yoda Admin Portal, is in contradiction with Yoda architecture. Furthermore, the frequency of using such a feature to change the look and feel of template pages is very low. The original request of maintaining the look and feel of landing pages has to do with organizational agreements within Surf.

So we change the scope of thi request. We still feel that we can help to improve this process by introducing the following:

YDA-7129 Configuration Playbook To Enforce Landing Template Check

YDA-7127 Preview Publication Landing Page

YDA-6852 Need to be able to customize publisher

These US are still under discussion. Unfortunately, I cannot grant view rights to these stories but I'll be happy to inform you about the specific scope of them.

Kind regads,

Orlando

**Comment by peer35** [comment]
@OrlandoYoDa thanks for picking this up again. 

I appreciate that managing the publication templates through the admin portal is contrary to the Yoda architecture because the publication portal is a separate thing. 

However, I would like to add that this part of the architecture works for the UU setup (multiple Yoda instances + 1 publication instance), but does not fit the SURF hosting setup (1 Yoda + 1 publication instance). At the VU we've had issues with the publication module after almost every Yoda upgrade since 2021 because of the way it's set up. 

Perhaps it's good to restate our requirements ahead of our discussion tomorrow:
- We are happy to use the same templates the UU is using, but **the VU is owner of the Yoda repository, publisher of the datasets contained in it and the contact point for any questions about them**. 
  This is an organizational choice the consortium made, but we can't just make SURF the owner, and certainly not the UU, that has legal implications.
- This must be clear on the landing pages by using branding (VU colors & logo are sufficient, no need to customize css) and "Vrije Universiteit Amsterdam" in text on the landing page (visible and in headers) and in the Publisher field of the DataCite metadata.
- There must be a contact link on the page for visitors who have a question which must lead to our own contact page.

So there are only a few parameters per SURF Yoda instance:
- Publisher/owner name
- Contact name
- Contact URL
- Brand color
- Brand logo

Furthermore:
- Like any other institute specific config, these parameters must persist over updates, even if the basic templates change.
- For managing the basic Yoda functionality an institutional admin should not need knowledge of coding, git, HTML or even icommands. 
  It would be nicest to be able to set the parameters ourselves in a friendly web interface, but we are also fine with sending our preferences to SURF so they can manage it for us. 

You're right these settings don't change very often. The trouble is that upgrades keep reverting the templates back to the UU defaults and it takes manual intervention, by us and/or SURF, to change them back. This shouldn't be difficult to change even in the current architecture.
